src/main/java/com/artipie/auth/AuthFromStorage.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.auth;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.asto.Key;
+import com.artipie.asto.blocking.BlockingStorage;
+import com.artipie.http.auth.Authentication;
+import com.jcabi.log.Logger;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.commons.codec.digest.DigestUtils;
+
+/**
+ * Authentication from abstract storage.
+ * The storage is expected to have yaml files with credentials in the following structure:
+ * <pre>
+ * ..
+ * ├── users
+ * │   ├── david.yaml
+ * │   ├── jane.yml
+ * │   ├── ...
+ * </pre>
+ * where the name of the file is username (case-sensitive), both yml and yaml extensions are
+ * supported. The yaml format file is the following:
+ * <pre>{@code
+ * david:
+ *   type: plain # plain and sha256 types are supported
+ *   pass: qwerty
+ *   email: david@example.com # Optional
+ *   enabled: true # optional default true
+ *   roles:
+ *     - java-dev
+ *   permissions:
+ *     artipie_basic_permission:
+ *       rpm-repo:
+ *         - read
+ * }</pre>
+ * @since 1.29
+ */
+public final class AuthFromStorage implements Authentication {
+
+    /**
+     * The storage to obtain users files from.
+     */
+    private final BlockingStorage asto;
+
+    /**
+     * Ctor.
+     * @param asto Abstract blocking storage
+     */
+    public AuthFromStorage(final BlockingStorage asto) {
+        this.asto = asto;
+    }
+
+    @Override
+    public Optional<User> user(final String name, final String pass) {
```

Comment:
As I understand, `User` class was removed from `http`.
Is this implementation temporary?

Reply:
hm, the class was renamed into `AuthUser` and moved to the upper level, as soon as http is released we will have to correct all the `Authentication` implementations.

Reply:
The corrections will be minimal, just change the class name